R/estimate_popsize.fit.R

```diff
@@ -0,0 +1,182 @@
+#' @title Regression fitting in single source capture-recapture models
+#' @author Piotr Chlebicki, Maciej Beręsewicz
+#'
+#' @description \code{estimate_popsize.fit} does for \code{estimate_popsize} what
+#' \code{glm.fit} does for \code{glm}. It is internally called in 
+#' \code{estimate_popsize}. Since \code{estimate_popsize} does much more than
+#' just regression fitting \code{estimate_popsize.fit} is much faster.
+#'
+#' @param y vector of dependent variables.
+#' @param X model matrix, the vglm one.
+#' @param family same as model in \code{estimate_popsize}.
+#' @param control control parameters created in \code{control.model}.
+#' @param method method of estimation same as in \code{estimate_popsize}.
+#' @param prior.weights vector of prior weights its the same argument as weights
+#' in \code{estimate_popsize}.
+#' @param start initial value of regression parameters.
+#' @param ... arguments to pass to other methods.
+#' 
+#' @details If \code{method} argument was set to \code{"mle"} the \code{stats::optim}
+#' function will be used to fit regression with analyticly computed gradient and 
+#' (minus) log likelihood functions as \code{gr} and \code{fn} arguments. 
+#' Unfortunately \code{optim} does not allow for hessian to be specified.
+#' More information about how to modify \code{optim} fitting is included in 
+#' [control.method()].
+#' 
+#' 
+#' If \code{method} argument was set to \code{"robust"} the iteratively reweighted 
```

Comment:
I am thinking about the `robust` name. At this stage we do not have a robust method to estimate the parameters so maybe a better way would be to change `robust` to `irls`?

Reply:
In this case we also ought to change `mle` to `optim`, point taken I'll correct it.